src/control/states/HoldingState.ts

```diff
@@ -36,17 +36,14 @@ class HoldingState extends State {
 
   public onChange(ctx: Parameters<State["onChange"]>[0]): void {
     const { flicking, axesEvent, transitTo } = ctx;
-
+    const animatingContext = flicking.control.controller.animatingContext;
     const inputEvent = axesEvent.inputEvent as { offsetX: number; offsetY: number };
-
-    const offset = flicking.horizontal
-      ? inputEvent.offsetX
-      : inputEvent.offsetY;
+    const offset = flicking.horizontal ? inputEvent?.offsetX : inputEvent?.offsetY;
 
     const moveStartEvent = new ComponentEvent(EVENTS.MOVE_START, {
       isTrusted: axesEvent.isTrusted,
       holding: this.holding,
-      direction: getDirection(0, -offset),
+      direction: inputEvent ? getDirection(0, -offset) : getDirection(animatingContext.start, animatingContext.end),
       axesEvent
```

Comment:
This will end up Flicking being in `DraggingState`, but it should actually be in `AnimatingState`.
You should rather change Flicking's state to `IdleState` before calling moveTo / moveToPanel.

Reply:
In the idle state, we change state to `AnimatingState` when onChange occurs.
Does this assume that there is no other way to fire change event in idle other than starting animation through the methods?

If so, it seems to be possible to set the `DraggingState` only when there is a native event as user input in the holding state, and if there is no native event, we can set it to `AnimatingState` like in the idle.
Can it be appropriate to define the next state like this?
`const nextState = inputEvent ? STATE_TYPE.DRAGGING : STATE_TYPE.ANIMATING;`

Reply:
The current state flow is:
- By user input: IDLE -> HOLDING -> DRAGGING -> ANIMATING -> IDLE (or IDLE -> HOLDING -> IDLE)
- By method call: IDLE -> ANIMATING -> IDLE

I think it's better to cancel the current input sequence on the method call than adding the new state flow like that.
Like, in the demo you provided(https://codepen.io/malangfox/pen/NWYvQKW) it triggers `Animation is already playing.` error if `moveTo` is called when Flicking's in `DraggingState`. fixing only the `HoldingState` doesn't fix that.

Also, as this fix doesn't care about previous inputs after changing it to `AnimatingState`, that will end up like this:
![1](https://user-images.githubusercontent.com/26213435/181146732-d933d37b-a5f8-4580-839b-fc815bda9258.gif)

You should cancel the previous inputs with Axes if the Flicking's `control.holding` is true, change the current state back to idle, and handle the moveTo animation like normal.